### PR TITLE
Race condition between device deferred removal and resume device

### PR DESCRIPTION
**Pull request by shishir-a412ed**
This PR will sync the "race condition" patch with upstream patch.

### CODE REVIEW

**Comment by shishir-a412ed** [comment]
ping @rhvgoyal

**Comment by runcom** [comment]
LGTM

Ping @rhvgoyal

File: daemon/graphdriver/devmapper/deviceset.go [hunk 1]
```diff
@@ -842,6 +842,55 @@ func (devices *DeviceSet) createRegisterDevice(hash string) (*devInfo, error) {
...
+
+	if err = devices.createRegisterSnapDevice(hash, baseInfo); err != nil {
+		return err
+	}
+
+	return nil
+}
+
 func (devices *DeviceSet) createRegisterSnapDevice(hash string, baseInfo *devInfo) error {
 	if err := devices.poolHasFreeSpace(); err != nil {
```
Comment (rhvgoyal) [inline daemon/graphdriver/devmapper/deviceset.go]: We moved poolHasFreeSpace() check in takeSnapshot() upstream. Why are we doing it in createRegisterSnapDevice() here? In fact, we seem to be doing it both in takeSnapshot() as well as createRegisterSnapDevice(). We need to remove this check from here.

Comment (shishir-a412ed) [inline daemon/graphdriver/devmapper/deviceset.go]: Fixed. PTAL.

**Comment by rhvgoyal** [comment]
LGTM. Thanks Shishir.

**Comment by runcom** [comment]
Merging I'll cherry pick in fedora-1.10.3 branch. Thanks all.

**Pull request merged**